scheduler/context.go

```diff
@@ -133,6 +133,13 @@ func (e *EvalContext) ProposedAllocs(nodeID string) ([]*structs.Allocation, erro
 	// Materialize the proposed slice
 	proposed = make([]*structs.Allocation, 0, len(proposedIDs))
 	for _, alloc := range proposedIDs {
+		if alloc.Job == nil {
```

Comment:
I think we just introduce a `SharedResources *Resource` to the Allocation. Then we can store things that apply to all tasks there. And then you can add it to the `Resources` in the FSM and planner can use it to. Lets avoid copying and placing the job in the `ProposedAllocs`. Less performant and will be hard to remember the subtly of this, making it brittle